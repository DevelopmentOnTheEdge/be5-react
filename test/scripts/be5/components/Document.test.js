import React          from 'react';
import renderer       from 'react-test-renderer';
import Document       from '../../../../src/scripts/be5/components/document';
import Form           from '../../../../src/scripts/be5/components/forms/form';
import Table           from '../../../../src/scripts/be5/components/tables/table';
import changeDocument from '../../../../src/scripts/be5/core/changeDocument';
import StaticPage     from '../../../../src/scripts/be5/components/staticPage';
import be5            from '../../../../src/scripts/be5/be5';

test('text', () => {
    const component = renderer.create(
        <Document />
    );

    expect(component.toJSON()).toMatchSnapshot();

    changeDocument('MainDocument', { component: 'text', value: "Page loaded." } );
    expect(component.toJSON()).toMatchSnapshot();
});

test('StaticPage', () => {
    const component = renderer.create(
        <Document />
    );

    changeDocument('MainDocument', { component: StaticPage,
            value: StaticPage.createValue("Test", 'test content')});
    expect(component.toJSON()).toMatchSnapshot();
});

test('Form', () => {
    const  component = renderer.create(
      <Document />
    );

    changeDocument('MainDocument', { component: Form,
            value: {
              "data": {
                "attributes": {
                  "bean": {
                    "values": {
                      "companyID": ""
                    },
                    "meta": {
                      "/companyID": {
                        "displayName": "companyID"
                      }
                    },
                    "order": [
                      "/companyID"
                    ]
                  },
                  "entity": "companies",
                  "layout": {},
                  "operation": "SelectCompany",
                  "operationResult": {
                    "status": "open"
                  },
                  "query": "Selection view SelectCompany",
                  "selectedRows": "",
                  "title": "SelectCompany"
                },
                "type": "form"
              },
              "links": {
                "self": "form/companies/Selection view SelectCompany/SelectCompany"
              },
              "meta": {
                "_ts_": "1507198229686"
              }
            }
    });
    expect(component.toJSON()).toMatchSnapshot();
});

test('Table', () => {
  const  component = renderer.create(
    <Document />
  );

  changeDocument('MainDocument', { component: Table,
    value: {
      "data": {
        "attributes": {
          "category": "companies",
          "columns": [
            "Наименование"
          ],
          "hasAggregate": false,
          "layout": {
            "type": "tableForm"
          },
          "length": 1,
          "operations": [
            {
              "clientSide": false,
              "isClientSide": false,
              "name": "Edit",
              "requiresConfirmation": false,
              "title": "Редактировать",
              "visibleWhen": "oneSelected"
            },
            {
              "clientSide": false,
              "isClientSide": false,
              "name": "Insert",
              "requiresConfirmation": false,
              "title": "Добавить",
              "visibleWhen": "always"
            }
          ],
          "page": "Общие сведения",
          "parameters": {},
          "rows": [
            {
              "cells": [
                {
                  "content": "adfd",
                  "options": {
                    "quick": {
                      "visible": "false"
                    }
                  }
                }
              ],
              "id": "19"
            }
          ],
          "selectable": true,
          "title": "Организации: Общие сведения",
          "totalNumberOfRows": 1
        },
        "type": "table"
      },
      "links": {
        "self": "table/companies/Общие сведения"
      },
      "meta": {
        "_ts_": "1507198487234"
      }
    }
  });
  expect(component.toJSON()).toMatchSnapshot();
});