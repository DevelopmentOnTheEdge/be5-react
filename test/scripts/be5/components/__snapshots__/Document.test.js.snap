// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form 1`] = `
<div
  className="document-content"
  id="document-content___MainDocument"
>
  <div
    className="row"
  >
    <div
      className="formBox col-xs-12 max-width-970 formBoxDefault"
    >
      <h1
        className="form-component__title"
      >
        SelectCompany
      </h1>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="property-set row"
        >
          <div
            className="form-group property col-lg-12 required"
          >
            <label
              className="form-control-label"
              htmlFor="companyIDField"
            >
              companyID
            </label>
            <div
              className="controls"
            >
              <input
                className="form-control"
                disabled={undefined}
                id="companyIDField"
                onChange={[Function]}
                placeholder={undefined}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="formActions"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Submit
          </button>
           
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`StaticPage 1`] = `
<div
  className="document-content"
  id="document-content___MainDocument"
>
  <div
    className="staticPage"
  >
    <div
      className="staticPage__text"
    >
      Test
    </div>
    <div
      className="staticPage__text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test content",
        }
      }
    />
  </div>
</div>
`;

exports[`Table 1`] = `
<div
  className="document-content"
  id="document-content___MainDocument"
>
  <div
    className="table-component"
  >
    <h1
      className="table-component__title"
    >
      Организации: Общие сведения
    </h1>
    <div>
      <div
        className="operationList"
      >
        <a
          className="btn btn-secondary btn-md"
          href=""
          onClick={[Function]}
        >
          Редактировать
        </a>
         
        <a
          className="btn btn-secondary btn-md"
          href=""
          onClick={[Function]}
        >
          Добавить
        </a>
      </div>
      <div
        id="quickColumns"
      >
        <span>
          Другие колонки:
        </span>
        <span>
          <input
            checked={false}
            id="quick0"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="rowIndex"
            htmlFor="quick0"
          >
            Наименование
             
          </label>
        </span>
      </div>
      <div
        className="scroll"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`text 1`] = `
<div
  className="document-content"
  id="document-content___MainDocument"
/>
`;

exports[`text 2`] = `
<div
  className="document-content"
  id="document-content___MainDocument"
>
  <h1>
    Page loaded.
  </h1>
</div>
`;
